app/src/main/java/com/keylesspalace/tusky/util/HttpHeaderLink.kt

```diff
@@ -0,0 +1,150 @@
+/* Written in 2017 by Andrew Dawson
+ *
+ * To the extent possible under law, the author(s) have dedicated all copyright and related and
+ * neighboring rights to this software to the public domain worldwide. This software is distributed
+ * without any warranty.
+ *
+ * You should have received a copy of the CC0 Public Domain Dedication along with this software.
+ * If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+package com.keylesspalace.tusky.util
+
+import android.net.Uri
+
+/**
+ * Represents one link and its parameters from the link header of an HTTP message.
+ *
+ * @see [RFC5988](https://tools.ietf.org/html/rfc5988)
+ */
+class HttpHeaderLink private constructor(uri: String) {
+    data class Parameter(var name: String? = null, var value: String? = null)
+
+    private val parameters: MutableList<Parameter> = ArrayList()
+
+    var uri: Uri
+
+    init {
+        this.uri = Uri.parse(uri)
+    }
+
+    private class ValueResult {
+        var value: String? = null
+            set(value) {
+                value ?: return
+                val v = value.trim()
+                if (v.isNotEmpty()) {
+                    field = v
+                }
+            }
+        var end: Int = -1
+    }
+
+    companion object {
+        private fun findEndOfQuotedString(line: String, start: Int): Int {
+            var i = start
+            while (i < line.length) {
+                val c = line[i]
+                if (c == '\\') {
+                    i += 1
+                } else if (c == '"') {
+                    return i
+                }
+                i++
+            }
+            return -1
+        }
+
+        private fun parseValue(line: String, start: Int): ValueResult {
+            val result = ValueResult()
+            val foundIndex = line.indexOfAny(charArrayOf(';', ',', '"'), start, false)
+            if (foundIndex == -1) {
+                result.value = line.substring(start)
+                return result
+            }
```

Comment:
How about making `value` a constructor param of `ValueResult` and then something like

```suggestion
            val foundIndex = line.indexOfAny(charArrayOf(';', ',', '"'), start, false)
            if (foundIndex == -1) {
                return ValueResult(line.substring(start))
            }
```